drivers/net/native_sim_adapt.c

```diff
@@ -0,0 +1,376 @@
+/**
+ * Copyright (c) 2023 Marcin Niestroj
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <netinet/in.h>
+#include <poll.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "native_sim.h"
+#include "native_sim_errno.h"
+
+#include <stdio.h>
+
+static int socket_family_from_nsim(int family, int *family_host)
+{
+	switch (family) {
+	case NSIM_AF_UNSPEC:
+		*family_host = AF_UNSPEC;
+		break;
+	case NSIM_AF_INET:
+		*family_host = AF_INET;
+		break;
+	default:
+		return -NSIM_EAFNOSUPPORT;
```

Comment:
printing an error here would help whoever hits this
Something like:
```
nsi_print_warning("%s: socket family %i not supported\n", __func__, family);
```

Same for the 2 functions below

Reply:
Added warnings, thanks for suggestion.